Source/InvocationCollection.cs

```diff
@@ -69,7 +69,7 @@ public int Count
 		{
 			get
 			{
-				lock (this.invocations)
+				lock (this.invocationsLock)
 				{
 					return this.invocations[index];
```

Comment:
Could you please add an explicit index-in-range check here / throw `IndexOutOfRangeException` otherwise?

Reply:
Won't we already just get one from the array anyway? Did you want to add some specific detail?

Reply:
There may be no array; you set the field to `null`. I believe that an out-of-range index access should reliably throw `IndexOutOfRangeException` instead of leaking implementation details.

Reply:
Ah, I hadn't thought of this scenario. I wouldn't want a `NullReferenceException` either.